database/database.go

```diff
@@ -186,3 +190,111 @@ func (db DatastoreBacked) StatesWithStatus(status PreloadStatus) (domains []Doma
 	return db.statesForQuery(
 		datastore.NewQuery("DomainState").Filter("Status =", string(status)))
 }
+
+// GetIneligibleDomain returns the state for the given domain.
+func (db DatastoreBacked) GetIneligibleDomains(domains []string) (states []IneligibleDomainState, err error) {
```

Comment:
The name of the function makes it sound like this returns a list of domains, but in reality this returns the states for the domains it is given, I think GetIneligibleDomainsState or something like that would probably be more descriptive.